### ISSUE TITLE
Support download Edge add-ons natively

### ISSUE THREAD

**Issue by Cyberavater** [kiwibrowser/src#188]
It'd be nice if Kiwi could natively download extensions from Edge Add-ons, as some of the extension is missing from Chrome Web Store.

A simple user-agent trick like Opera does not seem applicable here.

Eg link; https://microsoftedge.microsoft.com/addons/detail/universal-bypass/ckiidekccfgninkobmmofopbbdgdclgg

This issue was referenced in #145

**Comment by Delphox** [comment]
In the meantime I've created an userscript that creates a button on Edge add-on pages to download the CRX through Edge webstore's API URL that you can install directly to Kiwi or to your userscript manager of choice.

https://gist.github.com/Delphox/b5cae8fb55f901422391833a1ae4ae72/raw/99018c5c413ee42b11c4587e3ff3aba0add966af/kiwi_edge_addons.user.js

It's pretty dirty and you also might need to refresh the add-on page if you opened it from the add-on gallery in the same page due to it not handling dynamic loading properly (I'm not very good at this 😅), but should work fine otherwise.

**Comment by kiwibrowser** [comment]
@Delphox 😻

**Comment by kiwibrowser** [comment]
https://github.com/kiwibrowser/src/commit/15020cb2c8969fa02e60b146dad2fcd13f297e00

Let's see if this works

**Comment by Delphox** [comment]
Thanks a lot for adding it so quickly @kiwibrowser ! Really appreciate it!

I updated to playstore version Git201006 (got it earlier thanks to apk.support) however the Get button doesn't seem to be working. It does get regular blue upon loading an extension page (not when coming from the catalog though, same issue my userjs had), however the button doesn't do anything. Could you look into it?

**Comment by kiwibrowser** [comment]
Yeah you're right it's not working, I'll check again why. You can see I took a slightly different approach but at the end it should work. Let's see :)

**Comment by Delphox** [comment]
Just got home and decided to connect devtools here, turns out just removing the "disabled" attribute here works!
![image](https://user-images.githubusercontent.com/5883538/95362072-f0592e80-08a3-11eb-8426-9ab137c9b64c.png)
![image](https://user-images.githubusercontent.com/5883538/95362421-71182a80-08a4-11eb-8a66-0c1a14c0e5d5.png)


But in the meantime would be nice if you would be able to fix the buttons from lazy-loading (?) sections not updating. Including the dynamic ones when opening another extension from within the store.

**Comment by Delphox** [comment]
So I did some further digging and this is what triggers the banner:
![image](https://user-images.githubusercontent.com/5883538/95603589-dd259a80-0a2c-11eb-8316-28b2508b3b76.png)

So basically having `Edg/<number>.<number>` anywhere on the javascript useragent, regardless of the numbers or position (or even if the nav.ua is literally _just that_, even setting it to just `Edg/0.0` works) gets rid of the banner and doesn't leave a padding behind. So that's probably the easiest way.

**Comment by kiwibrowser** [comment]
Thanks for the research,

I did JS UA override in Google Chrome extension store here:
https://github.com/kiwibrowser/src/commit/71134fb5979e7900217c052b31d3859fffb859c0
if you need a reference, otherwise I can do it (but first I need to take care of the next release)

**Comment by Delphox** [comment]
Oh I'd appreciate if you did it. I'm not really good at this and can't build it locally to test. But take your time and thanks for what you're doing!

**Comment by kiwibrowser** [comment]
Testing @ https://github.com/kiwibrowser/src/commit/8aac563f925c9b538a9b14e7fb21003daf4abd0f

**Comment by Delphox** [comment]
Cool! Just so we're on the same page, the UA hint is just to remove the "install edge" banner. You still need to fit `individualButton.disabled=null` (?) somewhere to make it actually respond.

**Comment by kiwibrowser** [comment]
Ah right, I had individualButton.removeAttribute('disabled') in one of my test versions but I forgot to put it there. It's building right now. I have put window.open as well, as window.location.replace seems not to work in some cases due to https to http downgrade (during a redirect if you try to access a http page, in some cases the browser may refuses to navigate)

**Comment by Delphox** [comment]
Tested and it works great! Both the button and the lack of banner. Now the last piece of the puzzle would be making it still update the button of dynamically-loaded elements (when you navigate between extensions within the same page or when you scroll down on the main page and more extensions load). After that it's basically perfect! 😄
